shm/player.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include <memory>
+#include "cargo.hpp"
+#include "ship.hpp"
+
+class Player {
+public:
+    Player(Ship& ship, size_t money, size_t availableSpace);
```

Comment:
Passing ship as a reference and then making shared_ptr is a bad idea. Pass shared ptr instead.